This patch causes busybox-1.15.3 ash read command to behave exactly like bash or zsh read command. The problem is that as of now
it behaves neither like bash/zsh nor like ash on my ubuntu-9.04 system (which is a symlink to dash).

According to http://www.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_05 (provided I interpret it correctly)
the bash's behaviour is the correct one.

Some test cases:

 * printf '\t,\ta\t,\tb\tc' | $(ONE_OF_SHELLS_BELOW) -c 'IFS=$(printf " \t,") read a b c d; echo ".$a. .$b. .$c. .$d."'

   bash/zsh:                   .. .a. .b. .c.
   ash(dash):                  .a. .b. .c. ..
   busybox-1.15.3 ash:         .a. .b. .c. ..
   busybox with patch applied: .. .a. .b. .c.

 * printf '\t\ta\t,\tb\tc' | $(ONE_OF_SHELLS_BELOW) -c 'IFS=$(printf " \t,") read a b c d; echo ".$a. .$b. .$c. .$d."'

   bash/zsh:                   .a. .b. .c. ..
   ash(dash):                  .a. .b. .c. ..
   busybox-1.15.3 ash:         .a. .b. .c. ..
   busybox with patch applied: .a. .b. .c. ..

 * printf '\t,\ta\t,,\tb\tc' | $(ONE_OF_SHELLS_BELOW) -c 'IFS=$(printf " \t,") read a b c d e; echo ".$a. .$b. .$c. .$d. .$e."'

   bash/zsh:                   .. .a. .. .b. .c.
   ash(dash):                  .a. .b. .c. .. ..
   busybox-1.15.3 ash:         .a. .. .b. .c. ..
   busybox with patch applied: .. .a. .. .b. .c.

--- shell/ash.c	2009-12-12 22:16:38.000000000 +0100
+++ shell/ash.c	2010-01-04 00:00:00.000000000 +0100
@@ -12623,7 +12623,7 @@
 #endif
 
 	status = 0;
-	startword = 2;
+	startword = 1;
 	backslash = 0;
 #if ENABLE_ASH_READ_TIMEOUT
 	if (timeout) /* NB: ensuring end_ms is nonzero */
